src/openforms/js/components/admin/form_design/logic/LogicDescription.js

```diff
@@ -0,0 +1,77 @@
+import isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+import React, {useContext, useEffect, useRef, useState} from 'react';
+
+import {APIContext} from 'components/admin/form_design/Context';
+import {LOGIC_DESCRIPTION_ENDPOINT} from 'components/admin/form_design/constants';
+import {TextInput} from 'components/admin/forms/Inputs';
+import {post} from 'utils/fetch';
+
+const generateDescription = async (csrftoken, expression) => {
+  let response;
+  try {
+    response = await post(LOGIC_DESCRIPTION_ENDPOINT, csrftoken, {expression});
+  } catch (error) {
+    // we deliberately ignore backend errors here - validation errors for bad logic
+    // expressions should already show up in that field itself.
+    console.debug(error);
+    return null;
+  }
+  return response.data.description;
+};
+
+const LogicDescriptionInput = ({
+  generationAllowed,
+  generationRequest,
+  onChange: _onChange,
+  onDescriptionGenerated,
+  logicExpression = null,
+  ...textInputProps
+}) => {
+  const {csrftoken} = useContext(APIContext);
+  const [modifiedByHuman, setModifiedByHuman] = useState(false);
+  const [hasFocus, setHasFocus] = useState(false);
+
+  const onChange = event => {
+    const {
+      target: {value},
+    } = event;
+    _onChange(event);
+    setModifiedByHuman(value !== '');
+  };
+
+  const mustBail = isEmpty(logicExpression) || !generationAllowed || modifiedByHuman || hasFocus;
```

Comment:
Maybe if the expression is invalid JSON it should also bail?

Reply:
this should already the case as the `logicExpression` prop needs to be valid JSON - and it's only updated from the JSONWidget to the parent state when the JSON validates. But it's possible something broke there while I was trying to clean up the code. Is this a theoretical remark or did you see it happen? :P

Reply:
It was theoretical! :smile: